### PR TITLE
Go Gen

### PR DESCRIPTION
**Pull request by RishabhPatil**
* Use `make gen go ent` to generate go client files
* Moved generation, preprocessing and postprocessing scripts to `client/scripts/` directory
* Separate folders for python and golang
* Added `cluster_ping.go` example under `go/examples/`

### CODE REVIEW

File: pyclient/apps/go/cluster_ping.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
+package main
+
+
+import (
+	"github.com/pensando/psm-go/psm_ent"
```
Comment (jainvipin) [inline pyclient/apps/go/cluster_ping.go:5]: is psm-go a top level repo in pensando org? And is it expected to only have generated code? If unsure, for now can you point it to `psm-tools/psmclient/go/psm_ent` in this PR while we discuss the generated code like we did for pypi?

File: pyclient/apps/go/cluster_ping.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+)
+
+func main() {
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	cfg := psm_ent.NewConfiguration()
+	cfg.Scheme = "https"
+	client := psm_ent.NewAPIClient(cfg)
+	ClusterV1Api := (*psm_ent.ClusterV1ApiService)(&client.Common)
+	
+	ctx := context.Background()
```
Comment (jainvipin) [inline pyclient/apps/go/cluster_ping.go:19]: do w need to cancel this context with defer?

File: pyclient/apps/go/cluster_ping.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	// Cluster Uptime
+	currentTime  := cluster.Status.CurrentTime
+	creationTime := cluster.Meta.CreationTime
+	uptime := currentTime.Sub(*creationTime)
+	fmt.Println("Cluster Uptime: ", uptime)
+
+	nodes_unhealthy := false
+	for _, node := range *cluster.Status.QuorumStatus.Members {
+		conditions := *node.Conditions
+		node_health := conditions[0].Type
```
Comment (jainvipin) [inline pyclient/apps/go/cluster_ping.go:31]: can you assume condition [0] will always be status? This code needs a bit of work to loop through all conditions and look for the relevant ones maybe?

File: pyclient/apps/go/go.mod [hunk 4]
```diff
@@ -0,0 +1,18 @@
+module examples
```
Comment (jainvipin) [inline pyclient/apps/go/go.mod:1]: pyclient having `go` and `go.mod` seems a bit strange... can we organize the structure a bit better?
are these packages needed only for the apps? and nothing for the code generation?

File: pyclient/apps/utils/helper.py [hunk 5]
```diff
@@ -13,7 +13,7 @@ def import_lib(pipelines = ["ent", "cloud"]):
             import pensando_lib
             return pensando_lib
         except Exception as e:
-            # raise e
+            raise e
```
Comment (jainvipin) [inline pyclient/apps/utils/helper.py:16]: this change seem unrelated to the go generation? can we take it out and keep in a different PR, if needed?

**Review by jainvipin — CHANGES_REQUESTED** [review]
@RishabhPatil  - thanks for the PR; let's do an offline review to address some base comments and any restructuring that we may need as a result.

**Comment by jsilberm** [comment]
pyclient/gogen_ent.sh, pyclient/gogenconfig_ent.json :  Don't these should be included, since they're generated files

**Review by jainvipin — APPROVED** [review]
LGTM

**Pull request merged**